src/validationinterface.cpp

```diff
@@ -128,52 +132,69 @@ void SyncWithValidationInterfaceQueue() {
     promise.get_future().wait();
 }
 
-void CMainSignals::MempoolEntryRemoved(CTransactionRef ptx, MemPoolRemovalReason reason) {
-    if (reason != MemPoolRemovalReason::BLOCK && reason != MemPoolRemovalReason::CONFLICT) {
-        m_internals->m_schedulerClient.AddToProcessQueue([ptx, this] {
-            m_internals->TransactionRemovedFromMempool(ptx);
-        });
-    }
-}
-
 void CMainSignals::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlockIndex *pindexFork, bool fInitialDownload) {
     // Dependencies exist that require UpdatedBlockTip events to be delivered in the order in which
     // the chain actually updates. One way to ensure this is for the caller to invoke this signal
     // in the same critical section where the chain is updated
 
-    m_internals->m_schedulerClient.AddToProcessQueue([pindexNew, pindexFork, fInitialDownload, this] {
+    auto func = __func__;
+    m_internals->m_schedulerClient.AddToProcessQueue([func, pindexNew, pindexFork, fInitialDownload, this] {
+        LogPrint(BCLog::VALIDATIONINTERFACE, "%s: new block hash=%s fork block hash=%s initial download=%d\n", func,
+                 pindexNew->GetBlockHash().ToString(),
+                 pindexFork ? pindexFork->GetBlockHash().ToString() : "null",
+                 fInitialDownload);
         m_internals->UpdatedBlockTip(pindexNew, pindexFork, fInitialDownload);
     });
 }
 
 void CMainSignals::TransactionAddedToMempool(const CTransactionRef &ptx) {
-    m_internals->m_schedulerClient.AddToProcessQueue([ptx, this] {
+    auto func = __func__;
+    m_internals->m_schedulerClient.AddToProcessQueue([func, ptx, this] {
+        LogPrint(BCLog::VALIDATIONINTERFACE, "%s: tx hash=%s\n", func, ptx->GetHash().ToString());
```

Comment:
nit: call this the 'txid' instead of 'tx hash', and also log the wtxid (use `GetWitnessHash()`)